### PR TITLE
Update release notes for Azure Spring April release

### PR DESCRIPTION
**Pull request by yiliuTo**
This PR is to update April release notes for Azure Spring:
- Added installation instructions for Azure Spring users.
- Added release highlights for Azure Spring modules.

### CODE REVIEW

File: releases/2021-04/java.md [hunk 1]
```diff
@@ -1536,6 +1622,11 @@ String messageText = body.toString();
 ### azure-resourcemanager-azurestack 1.0.0-beta.1 [Changelog](https://github.com/Azure/azure-sdk-for-java/blob/azure-resourcemanager-azurestack_1.0.0-beta.1/sdk/azurestack/azure-resourcemanager-azurestack/CHANGELOG.md#100-beta1-2021-04-12)
 - Azure Resource Manager AzureStack client library for Java. This package contains Microsoft Azure SDK for AzureStack Management SDK. Azure Stack. Package tag package-preview-2020-06. For documentation on how to use this package, please see [Azure Management Libraries for Java](https://aka.ms/azsdk/java/mgmt).
 
+### Azure Spring Boot [Changelog](https://github.com/Azure/azure-sdk-for-java/blob/azure-spring-boot_3.4.0/sdk/spring/azure-spring-boot/CHANGELOG.md#340-2021-04-19)
```
Comment (saragluna) [inline releases/2021-04/java.md:1625]: Should we have a section for Azure Spring Cloud here?

File: releases/2021-04/java.md [hunk 2]
```diff
@@ -45,9 +45,7 @@ azure-communication-identity:1.0.0
 azure-resourcemanager-msi:2.3.0
 azure-resourcemanager-sql:2.3.0
 azure-resourcemanager-redis:2.3.0
-azure-spring-cloud-context:2.3.0
```
Comment (chenrujun) [inline releases/2021-04/java.md:48]: It's an error by automated job? Can we update the job? Or let the job-creator update the job.

File: releases/2021-04/java.md [hunk 3]
```diff
@@ -45,9 +45,7 @@ azure-communication-identity:1.0.0
...
 azure-resourcemanager-delegatednetwork:1.0.0-beta.1
-azure-spring-integration-storage-queue:2.3.0
```
Comment (chenrujun) [inline releases/2021-04/java.md:50]: Same here.

File: releases/2021-04/java.md (hunk 1 above)
Comment (yiliuTo) [reply releases/2021-04/java.md:1625]: We don't have release notes for Azure Spring Cloud this month, so need we put an section header here?

**Review by chenrujun — APPROVED** [review]

File: releases/2021-04/java.md (hunk 1 above)
Comment (yiliuTo) [reply releases/2021-04/java.md:1640]: Add a section for Azure Spring Cloud with app configuration release notes

File: releases/2021-04/java.md (hunk 2 above)
Comment (yiliuTo) [reply releases/2021-04/java.md:48]: Hi, @g2vinay @czubair Seems like there is an automatic job to collect release information and then add them in each month's release note. And we found the information for Azure Spring Boot and Azure Spring Cloud libraries are outdated, for example, all the content I deleted in this PR are packages we releases in March. Could you help to check the releated job? Thanks.

**Pull request merged**